### ISSUE TITLE
NeoVim pastes an error message when the current working directory no longer exists

### ISSUE THREAD

**Issue by xanderdunn** [neovim/neovim#1821]
Mac OSX 10.10.1

Pasting text into an nvim session whose current working directory no longer exists will also paste in an error message.

Reproduce: 
1. mkdir testing
2. cd testing
3. nvim
4. :e test
5. In another terminal: rm testing
6. Back in the nvim terminal: Paste in some text

Result: NeoVim will paste in an error message, followed by the desired text:
`shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory`

This does _not_ occur on vim 7.4.488.

nvim --version

```
NVIM 0.0.0-alpha+201501130938 (compiled Jan 14 2015 12:56:29)
Commit: 9b4f6fbd33ebd452d472b0333accfcb34e01173b
Build type: RelWithDebInfo
Compilation: /usr/local/Library/ENV/4.3/clang -Wconversion -O2 -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -I/tmp/neovim-HJ2cLO/build/config -I/tmp/neovim-HJ2cLO/src -I/tmp/neovim-HJ2cLO/.deps/usr/include -I/tmp/neovim-HJ2cLO/.deps/usr/include -I/tmp/neovim-HJ2cLO/.deps/usr/include/luajit-2.0 -I/usr/local/opt/gettext/include -I/usr/include -I/tmp/neovim-HJ2cLO/build/src/nvim/auto -I/tmp/neovim-HJ2cLO/build/include
Compiled by admin@Alex-Dunns-Mac-Pro.local  Features included (+) or not (-):
+acl             +farsi           -mouse_gpm       -tag_any_white
+arabic          +file_in_path    +mouse_netterm   +terminfo
+autocmd         +find_in_path    +mouse_sgr       +termresponse
-browse          +float           -mouse_sysmouse  +textobjects
+byte_offset     +folding         +mouse_urxvt     +title
+cindent         -footer          +mouse_xterm     -toolbar
-clipboard       +fork()          +multi_byte      +user_commands
+cmdline_compl   +gettext         +multi_lang      +vertsplit
+cmdline_hist    +iconv           +path_extra      +virtualedit
+cmdline_info    +insert_expand   +persistent_undo +visual
+comments        +jumplist        +postscript      +visualextra
+conceal         +keymap          +printer         +viminfo
+cscope          +langmap         +profile         +vreplace
+cursorbind      +libcall         +quickfix        +wildignore
+cursorshape     +linebreak       +reltime         +wildmenu
+dialog_con      +lispindent      +rightleft       +windows
+diff            +listcmds        +scrollbind      +writebackup
+digraphs        +localmap        +signs           -X11
-dnd             +menu            +smartindent     -xfontset
-ebcdic          +mksession       +startuptime     -xsmp
-emacs_tags      +modify_fname    +statusline      -xterm_clipboard
+eval            +mouse           +syntax          
+ex_extra        -mouseshape      +tag_binary      
+extra_search    +mouse_dec       +tag_old_static  
   system vimrc file: "$VIM/nvimrc"
     user vimrc file: "$HOME/.nvimrc"
 2nd user vimrc file: "~/.nvim/nvimrc"
      user exrc file: "$HOME/.exrc"
  fall-back for $VIM: "/usr/local/Cellar/neovim/HEAD/share/nvim"

```

**Comment by justinmk** [comment]
Does this happen with clipboard disabled?

**Comment by alexdunn** [comment]
@justinmk Yes, this is the case. I disabled `set clipboard=unnamed` in my .vimrc and tested again.  All new yanks were pasted correctly, without the error message.  

It is, however, interesting to note that existing yanks remained tainted.  That is, any yanks I had made while `clipboard=unnamed` still included the error text when pasted in, even after I had turned off the clipboard setting.  

Thanks!
